examples/__init__.py

```diff
@@ -48,6 +49,10 @@ def transform(project: MkDocsConfig, config: MkDocsConfig):
     # Inherit settings for copyright
     project.copyright = config.copyright
 
+    # check that a theme has been defined and we are not running with the default theme
+    if project.theme == "mkdocs":
```

Comment:
I'm pretty sure this must be [`project.theme.name`](https://www.mkdocs.org/user-guide/choosing-your-theme/). Other than that, I'd check for `project.theme.name == "material"`, as this will also trigger for other themes than `mkdocs`.

On a side note, please make sure to always start comments and commit messages with a capital letter, i.e., "Check", instead of "check". We do this across the entirety of our code base. If it's only one sentence, don't terminate with a `.`. If it's multiple sentences, terminate them with `.`.

Reply:
Unfortunately, no. If no theme is defined, them `project.theme` is not an object but a string (Python, yay!). I bet the same is true if you just write `theme: material` - [as the MkDocs documentation suggests](https://www.mkdocs.org/getting-started/#theming-our-documentation). 

I agree about checking for anything that is not Material. I'd suggest throwing an error in such a case but if the default theme is detected I just substitute. The reason is that I want to have a `mkdocs serve` running all the time and be able to create new examples. Another way of doing this, of course, would be to not use `mkdocs new` but to create a template example that already has Material in it, then `cp -r`.

Aye cap'n re the comments.

Reply:
> I bet the same is true if you just write theme: material - [as the MkDocs documentation suggests](https://www.mkdocs.org/getting-started/#theming-our-documentation).

I was wondering why I see people keep doing that, and wasn't aware this is mentioned in the documentation. I think we should raise this to the maintainers of MkDocs, because two prominent pages pages offer different information how to configure the exact same thing:

- https://www.mkdocs.org/getting-started/#theming-our-documentation (`theme`)
- https://www.mkdocs.org/user-guide/choosing-your-theme/ (`theme.name`)

IMHO this is an inconsistency. Could you raise an issue at MkDocs?

Reply:
Re the comment below - how do I ignore an example? Would that be a modification to the projects plugin?

Reply:
Yes, will raise the issue with MkDocs.

Reply:
Ah, I wrote before I thought haha. Hmm, yeah, that's currently not supported. If your file has a `mkdocs.yml`, it will be included. We could add `_include` and `_exclude` for this, like for many of the other plugins, e.g. the [`optimize` plugin](https://squidfunk.github.io/mkdocs-material/plugins/optimize/#config.optimize_include), but I'm not sure those would be changable from the custom transform function (might be too late). Can you open a change request on the Material for MkDocs repository for ignoring examples? Or what's your opinion working with the plugin, would that make sense?

Reply:
Let me try and put in a nutshell what we are dealing with, in the most general terms. When using the project plugin, people may well write transform functions that try to pull in configuration elements that do not exist if Material is not the theme. 

This can be the case temporarily when people have just run `mkdocs new` or it could be deliberate. One crazy idea I just had was to write examples for both Material and Read the Docs using the projects plugin. In that case, people would need to implement suitable logic in their `transform` function. For the case that it is temporary, the MkDocs issue should do the trick, if they pick it up and ensure that `project.theme` is always an object.

Reply:
Ok, issue filed - hope it fits the bill: https://github.com/mkdocs/mkdocs/issues/3436

Reply:
Thanks! Issue LGTM. Regarding usage patterns: you#re gathering very valuable insights by using the plugin, and we should definitely put a large focus on making the usage as smooth as possible, i.e., dealing with incomplete configuration, including/excluding projects etc. I encourage you to document every hiccup, so we can use it to improve the plugin in the coming weeks.

Let's see what the MkDocs maintainers say in regard to standardizing configuration. I think `theme: str` is a relict from when MkDocs did not support custom themes.